tfjs-backend-webgpu/src/kernels/Draw.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use backend file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {KernelConfig, KernelFunc, TensorInfo} from '@tensorflow/tfjs-core';
+import {Draw, DrawAttrs, DrawInputs,} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {ToPixelsProgram} from '../to_pixels_webgpu';
+
+import {DrawTextureGraphicsProgram} from '../webgpu_graphics_program';
+
+export function draw(
+    args: {inputs: DrawInputs, backend: WebGPUBackend, attrs: DrawAttrs}):
+    TensorInfo {
+  const {inputs, backend, attrs} = args;
+  const {image} = inputs;
+  const {canvas, options} = attrs;
+  const [height, width] = image.shape.slice(0, 2);
+  const {imageOptions} = options || {};
+  const alpha = imageOptions ?.alpha || 1;
+
+  const format = 'rgba8unorm';
+  const outShape = [height, width];
+  const useAlpha = alpha !== 1;
+  const program = new ToPixelsProgram(outShape, image.dtype, format, useAlpha);
+  canvas.width = width;
+  canvas.height = height;
+  const gpuContext = canvas.getContext('webgpu');
+  if (!gpuContext) {
+    throw new Error(
+        `Please make sure this canvas has only been used for webgpu context!`);
+  }
+  const numChannels = image.shape.length === 3 ? image.shape[2] : 1;
+  const alphaMode = numChannels === 4 ? 'premultiplied' : 'opaque';
```

Comment:
Please always use `premultiplied`. Why use `opaque` when `numChannels !== 4`?

Reply:
For case like:
```
    const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
    const width = 6;
    const height = 2;
    const img = tf.tensor3d(data, [width, height, 1], 'int32');
    const drawOptions = {
      contextOptions: {contextType: env.name},
      imageOptions: {alpha: 0.5}
    };
```
This img is opaque,  single channel. 
If opaque mode, the readback data is: 
```
      1, 1, 1, 255, 1, 1, 1, 255, 2, 2, 2, 255, 2, 2, 2, 255,
      3, 3, 3, 255, 3, 3, 3, 255, 4, 4, 4, 255, 4, 4, 4, 255,
      5, 5, 5, 255, 5, 5, 5, 255, 6, 6, 6, 255, 6, 6, 6, 255
```
If premultiplied mode, the readback data is:
```
      2,  2,  2,  128, 2,  2,  2,  128, 4,  4,  4,  128, 4,  4,  4,  128,
      6,  6,  6,  128, 6,  6,  6,  128, 8,  8,  8,  128, 8,  8,  8,  128,
      10, 10, 10, 128, 10, 10, 10, 128, 12, 12, 12, 128, 12, 12, 12, 128
```
They are different. Based on the imput img is opaque, I think the result of 'opaque mode' is better.